api/v1alpha1/nicfirmwaresource_types.go

```diff
@@ -0,0 +1,60 @@
+/*
+2025 NVIDIA CORPORATION & AFFILIATES
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// NicFirmwareSourceSpec represents a list of url sources for FW
+type NicFirmwareSourceSpec struct {
+	// BinUrlSources represents a list of url sources for FW
+	BinUrlSources []string `json:"binUrlSources"`
+}
+
+// NicFirmwareSourceStatus represents the status of the FW from given sources, e.g. version available for PSIDs
+type NicFirmwareSourceStatus struct {
+	// State represents the firmware processing state, e.g. Downloading|Processing|Success|ProcessingFailed|DownloadFailed
+	State string `json:"state"`
```

Comment:
do we want to add enum validation for this field?

Reply:
Added enum validation